app/service/contact_svc.py

```diff
@@ -71,7 +71,8 @@ async def _save(self, result):
                 if result.output:
                     link.output = True
                     self.get_service('file_svc').write_result_file(result.id, result.output)
-                    operation = await self.get_service('data_svc').locate('operations', dict(id=link.operation))
+                    all_operations = await self.get_service('data_svc').locate('operations')
+                    operation = next((o for o in all_operations if any(lnk.id == result.id for lnk in o.chain)), None)
```

Comment:
Optimization-wise, we probably only need to pull from `running` operations right?